.github/workflows/build_push_image.yaml

```diff
@@ -0,0 +1,60 @@
+name: Build container image and maybe push it to the registry
```

Comment:
_Maybe_ because this is a reusable workflow and its behavior changes based on the input, so if the caller workflow (e.g. a CI) calls it passing the value `false` on the `push` variable, the workflow just builds the container image without pushing it to the container registry. If the caller sends `true` for the same variable, then the workflow also pushes the built container image to the container registry.

Reply:
I'd write (instead of "maybe" that sounds like this pipeline has some indeterministic behavior) "according to the value of push parameter"